Source/WebKit/WPEPlatform/wpe/wayland/WPEWaylandSeat.cpp

```diff
@@ -166,7 +166,12 @@ const struct wl_pointer_listener WaylandSeat::s_pointerListener = {
 
         seat.m_pointer.time = time;
 
-        GRefPtr<WPEView> view = wpeToplevelWaylandGetVisibleViewUnderPointer(seat.m_pointer.toplevel.get());
+        auto* toplevelWayland = seat.m_pointer.toplevel.get();
+        wpeToplevelWaylandSetIsFocused(toplevelWayland, true);
+        if (GRefPtr<WPEView> focusedView = wpeToplevelWaylandGetVisibleFocusedView(toplevelWayland))
+            wpe_view_focus_in(focusedView.get());
+
+        GRefPtr<WPEView> view = wpeToplevelWaylandGetVisibleViewUnderPointer(toplevelWayland);
```

Comment:
I think the focus on button press/touch should not be platform specific, it should be done by WebKit in WPEWebViewPlatform like gtk does in WebKitWebViewBase.